common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -51,6 +51,12 @@
 
     // The channel associated to the stream
     final Channel associatedChannel;
+    // Indicates whether the buffers is materialized, some buffers like
+    // ShuffleManagedBufferIterator, ShuffleChunkManagedBufferIterator, ManagedBufferIterator are
+    // materialized only when the iterator is requested by the next method, so we should not
+    // request and release them, to reduce I/O operations and improve performance for the External
+    // Shuffle Service.
```

Comment:
```suggestion
    // Indicates whether the buffers is only materialized when next() is called. Some buffers like
    // ShuffleManagedBufferIterator, ShuffleChunkManagedBufferIterator, ManagedBufferIterator are
    // not materialized until the iterator is traversed by calling next(). We use it to decide whether buffers
    // should be released at connectionTerminated() in order to avoid unnecessary buffer materialization,
   // which could be I/O based.
```